project_earned_value_task/models/project.py

```diff
@@ -0,0 +1,472 @@
+        res = cr.fetchone()
+        if res:
+            return res[0] or 0.0
+        else:
+            return 0.0
+
+    def _get_plan_cost_to_date(self):
+        cr = self.env.cr
+        today = datetime.today().strftime('%Y-%m-%d %H:%M:%S')
+        # Compute the Budget at Completion
+        cr.execute("""
+        SELECT SUM(PT.planned_hours * ip.value_float)
+        FROM project_task as PT
+        INNER JOIN resource_resource RES
+        ON RES.user_id = PT.user_id
+        INNER JOIN hr_employee EMP
+        ON EMP.resource_id = RES.id
+        INNER JOIN product_product PR
+        ON PR.id = EMP.product_id
+        INNER JOIN product_template PRT
+        ON PRT.id = PR.product_tmpl_id
+        LEFT JOIN ir_property ip
+        ON (ip.name='standard_price'
+        AND ip.res_id=CONCAT('product.template,',PRT.id)
+        AND ip.company_id=PT.company_id)
+        WHERE PT.project_id = %s
+        AND PT.date_end <= %s""", (self.id, today))
+        res = cr.fetchone()
+        if res:
+            return res[0] or 0.0
+        else:
+            return 0.0
+
+    def _get_actual_cost_to_date(self):
+        cr = self.env.cr
+        # Compute the actual cost
+        cr.execute("""
+        SELECT SUM(PTW.hours * ip.value_float)
+        FROM project_task_work AS PTW
+        INNER JOIN project_task AS PT
+        ON PTW.task_id = PT.id
+        INNER JOIN resource_resource RES
+        ON RES.user_id = PT.user_id
+        INNER JOIN hr_employee EMP
+        ON EMP.resource_id = RES.id
+        INNER JOIN product_product PR
+        ON PR.id = EMP.product_id
+        INNER JOIN product_template PRT
+        ON PRT.id = PR.product_tmpl_id
+        LEFT JOIN ir_property ip
+        ON (ip.name='standard_price'
+        AND ip.res_id=CONCAT('product.template,',PRT.id)
+        AND ip.company_id=PT.company_id)
+        WHERE PT.project_id = %s""", (self.id, ))
+        res = cr.fetchone()
+        if res:
+            return res[0] or 0.0
+        else:
+            return 0.0
+
+    def _get_earned_value_to_date(self):
+        cr = self.env.cr
+        # Compute the earned value
+        cr.execute("""
+        SELECT SUM(PT.planned_hours * ip.value_float * PTT.poc/100)
+        FROM project_task as PT
+        INNER JOIN project_task_type as PTT
+        ON PTT.id = PT.stage_id
+        INNER JOIN resource_resource RES
+        ON RES.user_id = PT.user_id
+        INNER JOIN hr_employee EMP
+        ON EMP.resource_id = RES.id
+        INNER JOIN product_product PR
+        ON PR.id = EMP.product_id
+        INNER JOIN product_template PRT
+        ON PRT.id = PR.product_tmpl_id
+        LEFT JOIN ir_property ip
+        ON (ip.name='standard_price'
+        AND ip.res_id=CONCAT('product.template,',PRT.id)
+        AND ip.company_id=PT.company_id)
+        WHERE PT.project_id = %s""", (self.id, ))
+        res = cr.fetchone()
+        if res:
+            return res[0] or 0.0
+        else:
+            return 0.0
+
+    @api.one
```

Comment:
@dreispt if deprecated, better find a better alternative. isnot there anything to improve this?
We need to think ahead about migration.